website_rate_product/models/mail_message.py

```diff
@@ -45,16 +45,19 @@ class ProductTemplate(osv.Model):
     _inherit = 'product.template'
 
     def _get_rating(self, cr, uid, ids, field_name, arg, context):
+        """
+        This method gets the rating for each rated template based on the
+        comments, the rating per comment is stored in the mail.message model
+        """
         res = {}
-        total = 0
-        cr.commit()
-        for pid in ids:
-            cr.execute("select avg(rating) from mail_message where res_id = %s\
-                        and model = 'product.template' and rating > 0;" %
-                       (pid))
-            fall = cr.fetchall()
-            total = fall[0][0] if fall else total
-            res[pid] = total
+        str_tuple = str(tuple(ids)).replace(',)', ')')
+        cr.execute("""
+          select avg(rating), res_id from mail_message where res_id in %s\
+          and model = 'product.template' and rating > 0 group by res_id;
+          """ % (str_tuple))
+        for record in cr.fetchall():
+            total = record[0]
```

Comment:
I don't understand the focus of `total` variable.
Could you help me to understand it?

What about use:
```python
for rating, product_template_id in cr.fetchall():
   res[product_template_id] = rating
```